tools/dynamo/gb_id_mapping.py

```diff
@@ -8,42 +8,60 @@
 
 
 def get_source_segment(source, node):
-    if hasattr(ast, "get_source_segment"):
-        return ast.get_source_segment(source, node)
+    return ast.get_source_segment(source, node)
 
 
-"""
-Normalizes string literals by removing formatting artifacts and escape sequences.
-Handles f-strings, quotes, newlines, and other syntax elements for cleaner output.
-"""
+def load_registry(path):
+    if path.exists():
+        with path.open() as f:
+            return json.load(f)
+    return {}
+
+
+def save_registry(reg, path):
+    with path.open("w") as f:
+        json.dump(reg, f, indent=2)
+
+
+def next_gb_id(reg):
+    ids = [int(x[2:]) for x in reg if x.startswith("GB") and x[2:].isdigit()]
+    return f"GB{(max(ids, default=0) + 1):04d}"
 
 
 def clean_string(s):
-    if s is None:
-        return None
+    """
+    Normalizes string literals by removing formatting artifacts and escape sequences.
+    Handles f-strings, quotes, newlines, and other syntax elements for cleaner output.
+    """
     if isinstance(s, str):
+        # Convert f-string prefix to regular string prefix (e.g., f"hello" -> "hello")
         s = re.sub(r'^f["\']', r'"', s)
+        # Replace quoted strings with f-prefix in the middle with a space (e.g., " f"" -> " ")
         s = re.sub(r'["\'] f["\']', " ", s)
+        # Remove surrounding quotes, keeping only the content (e.g., "hello" -> hello)
         s = re.sub(r'^["\'](.*)["\']$', r"\1", s)
+        # Replace any whitespace around newlines with a single space for consistent formatting
         s = re.sub(r"\s*\n\s*", " ", s)
+        # Replace escaped quotes with their unescaped versions (e.g., \" -> ", \' -> ')
         s = s.replace('\\"', '"').replace("\\'", "'")
+        # Remove any remaining backslashes used for escaping
         s = s.replace("\\", "")
-        s = s.replace("\\", "")
+        # Replace adjacent quoted strings with a space (e.g., " "" -> " ")
         s = re.sub(r'" "', " ", s)
+        # Remove any curly brace expressions used in f-strings (e.g., {variable})
```

Comment:
I think we should keep the curly braces so the user is aware of what the string is supposed to show.